forte/data/base_pack.py

```diff
@@ -594,6 +640,18 @@ def entry_setter(cls: Entry, value: Any, attr_name: str):
                 tid=cls.tid, attr_name=attr_name, attr_value=attr_value
             )
 
+        # If this is the first time an entry of this type is
+        # created, its attributes do not have a getter and setter
+        # property associated with them. In this case, we manually
+        # add all dataclass attributes to the cached_attribute_data
+        # dict.
+        for name in entry.__dataclass_fields__:
+            if (
+                hasattr(type(entry), name)
+                and not isinstance(getattr(type(entry), name), property)
+            ) or (not hasattr(type(entry), name)):
+                entry.cached_attribute_data[name] = getattr(entry, name, None)
```

Comment:
So one workaround that I have in mind for this is to fetch the `_type_attributes` dict somehow to check if the current entry being created is present in `type_attributes` or no. In order to do that, I will have to add a method in `DataStore` to fetch the `_type_attributes`. Would that be okay?

Reply:
fetching `_type_attributes` should be fine. what's the difference in terms of the cost here?

further, how often would each of these boolean statements being invoked?

Reply:
So, checking if an entry is present in `type_attributes` would be O(1) time. 

- If an entry is present in `type_attributes`, we can assume that a getter and setter method is associated with its `dataclass` fields which in turn means that `cached_attributes` dict is already populated (so nothing needs to be done)
- If an entry is not present in `_type_attributes`, `cached_attributes` will need to be populated. We do this by iterating over all its `dataclass` attributes.

I feel that by introducing this check, we can get rid of the current `if` statement which you said is computationally expensive. The only thing we will have to check is if this way still works when we parse ontology files (which I feel it would).

Reply:
checking `type_attributes` will be preferable to doing class conversion, type checking etc.